### PR TITLE
Android Emulator now checks whether the SDK is correctly configured

### PR DESCRIPTION
**Pull request by teobugslayer**
Also, gives a proper message to the user

Implements http://teampulse.telerik.com/view#item/279833

### CODE REVIEW

File: mobile/android/android-emulator-services.ts [hunk 1]
```diff
@@ -32,6 +32,20 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 		iconv.extendNodeEncodings();
 	}
 
+	public checkDependencies(): IFuture<void> {
+		var future = new Future<void>();
+		var proc = this.$childProcess.spawn('emulator', ['-help']);
+		proc.on("error", (args:any) => {
+			this.$errors.fail("The Android SDK is not configured properly. " +
```
Comment (rosen-vladimirov) [inline mobile/android/android-emulator-services.ts]: or maybe future.throw(...) not sure which one's better

Comment (teobugslayer) [inline mobile/android/android-emulator-services.ts]: They have the same outcome.

File: mobile/android/android-emulator-services.ts [hunk 2]
```diff
@@ -32,6 +32,20 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
...
+				"Verify that you have installed the Android SDK and that you have added its `platform-tools` and `tools` directories to your PATH environment variable.");
+		});
+		proc.on("exit", (code:number) => {
+			future.return();
```
Comment (rosen-vladimirov) [inline mobile/android/android-emulator-services.ts]: is it possible to have both error and exit events, we stick to the following code:
```Typescript
if (!future.isResolved()) { 
    future.return();
}
```
Same is valid for on error callback

Comment (rosen-vladimirov) [inline mobile/android/android-emulator-services.ts]: From nodejs documentation:
```
Note that the exit-event may or may not fire after an error has occured. If you are listening on both events to fire a function, remember to guard against calling your function twice.
```
http://nodejs.org/api/child_process.html#child_process_event_error

**Comment by rosen-vladimirov** [comment]
:+1:

**Pull request merged**